src/libraries/System.Runtime.Loader/tests/GenerateHotReloadDelta.targets

```diff
@@ -0,0 +1,45 @@
+<Project>
+  <!-- Invoke hotreload-delta-gen to apply the DeltaScript to the output assembly to
+       create the .dmeta, .dil and .dpdb files on the OutputPath -->
+  <!-- Have to be careful about AfterTargets. If we happen to go after a design-time target, we could get into an infinite loop. -->
+  <Target Name="CompileDiff" AfterTargets="Build" Condition="'$(DeltaScript)' != ''">
+    <PropertyGroup>
+      <HotreloadDeltaGenCommand>"$(DotNetTool)" hotreload-delta-gen</HotreloadDeltaGenCommand>
```

Comment:
Presumably this adds a source build dependency onto hotreload-delta-gen. Is that repo sourcebuild compatible? Today we don't use any dotnet (repo) tools as part of the product build and I would like to better understand why this can't be a normal nuget package instead? Additional thoughts on this:

1. This adds an out-of-proc invocation which slows down the build
2. Using a dotnet tool requires them (the manifest) to be restored upfront. This works against the goal to make any project in the repository invokable without any up-front steps. If this would be a nuget package with a targets file in it instead, we would have a strong dependency and the package would only be restored if this project is restored.
3. When building other parts of the repository, this tool always needs to be restored.

If we control the tool ourselves then I would prefer to consume a nuget package with an msbuild targets file in it which then invokes a task to avoid the out-of-proc hop. Is that possible?

cc @ericstj

Reply:
> an msbuild targets file in it which then invokes a task to avoid the out-of-proc hop. Is that possible

This won't work - hotreload-delta-gen needs to open the current project as a roslyn MSBuildWorkspace - which throws an exception because the current project already has MSBuild loaded.

Reply:
If out-of-process isn't negotiable then I'm ok with that. Please consider my other suggestions as I believe a dependency on a dotnet tool isn't great.

Reply:
> If out-of-process isn't negotiable then I'm ok with that. Please consider my other suggestions as I believe a dependency on a dotnet tool isn't great.

@ViktorHofer okay, I'll give it a try.  Is this blog post a good guide to how to package a build tool in a nuget? https://natemcmaster.com/blog/2017/11/11/build-tools-in-nuget/  Is there anything special I need to do differnetly for our repos?

Reply:
Some of what's said in this blog doesn't apply anymore but in general this is a good guidance, yes. https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.ApiCompat/src/Microsoft.DotNet.ApiCompat.csproj is an example of how you would package such a tool in Arcade which handles most of what you need via the BuildTask.targets file which is imported in the project.

In general creating a nuget package which invokes an executable via a targets file underneath isn't too hard. Give it a try and I'm happy to review your changes where-ever they need to be done.

Reply:
Done.  Thanks for your help @ViktorHofer - I switched this PR over to using the Generator.BuildTool nupkg and removed the tool from the manifest.